Index: cryptsetup/tests/compat-test
===================================================================
--- cryptsetup.orig/tests/compat-test
+++ cryptsetup/tests/compat-test
@@ -46,6 +46,9 @@ TEST_UUID="12345678-1234-1234-1234-12345
 
 LOOPDEV=$(losetup -f 2>/dev/null)
 FIPS_MODE=$(cat /proc/sys/crypto/fips_enabled 2>/dev/null)
+# Circumvent test failure due to Bionic builder; we need to decrease
+# the memlock limit here to mimic Xenial builder (see LP #1891473).
+ulimit -l 0
 
 function remove_mapping()
 {
Index: cryptsetup-2.3.3/tests/luks2-validation-test
===================================================================
--- cryptsetup-2.3.3.orig/tests/luks2-validation-test
+++ cryptsetup-2.3.3/tests/luks2-validation-test
@@ -21,6 +21,10 @@ FAILS=0
 
 [ -z "$srcdir" ] && srcdir="."
 
+# Circumvent test failure due to Bionic builder; we need to decrease
+# the memlock limit here to mimic Xenial builder (see LP #1891473).
+ulimit -l 0
+
 function remove_mapping()
 {
 	rm -rf $IMG $TST_IMGS >/dev/null 2>&1
Index: cryptsetup-2.3.3/tests/tcrypt-compat-test
===================================================================
--- cryptsetup-2.3.3.orig/tests/tcrypt-compat-test
+++ cryptsetup-2.3.3/tests/tcrypt-compat-test
@@ -13,6 +13,10 @@ PIM=1234
 
 [ -z "$srcdir" ] && srcdir="."
 
+# Circumvent test failure due to Bionic builder; we need to decrease
+# the memlock limit here to mimic Xenial builder (see LP #1891473).
+ulimit -l 0
+
 function remove_mapping()
 {
 	[ -b /dev/mapper/$MAP ] && dmsetup remove --retry $MAP


